### PR TITLE
added unit tests and some code changes

### PR DESCRIPTION
**Pull request by saludangelito**

### CODE REVIEW

File: src/steps/users/contact-field-equals.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export class ContactFieldEquals extends BaseStep implements StepInterface {
   {
     field: 'operator',
     type: FieldDefinition.Type.STRING,
+    optionality: FieldDefinition.Optionality.OPTIONAL,
```
Comment (iamEAP) [inline src/steps/users/contact-field-equals.ts]: - [ ] Recommend maintaining the `REQUIRED` aspect here.  The only reason we made this optional in other Cogs was to maintain backward compatibility for existing steps.

File: src/steps/users/contact-field-equals.ts [hunk 2]
```diff
@@ -42,23 +43,18 @@ export class ContactFieldEquals extends BaseStep implements StepInterface {
     const stepData: any = step.getData().toJavaScript();
     const email: string = stepData.email;
     const field: string = stepData.field;
-    const operator: string = stepData.operator;
+    const operator: string = stepData.operator || 'be';
```
Comment (iamEAP) [inline src/steps/users/contact-field-equals.ts]: - [ ] Same.  No need to supply a default (see above)

File: src/steps/users/contact-field-equals.ts [hunk 3]
```diff
@@ -42,23 +43,18 @@ export class ContactFieldEquals extends BaseStep implements StepInterface {
...
       apiRes = await this.client.getContactByEmail(email);
-    } catch (e) {
-      return this.error('There was a problem checking the Contact: %s', [e.toString()]);
-    }
-
-    try {
       if (!apiRes.user) {
         // If no results were found, return an error.
         return this.error('No contact found for email %s', [email]);
-      } else if (!apiRes.user.hasOwnProperty(field) && !apiRes.user.dataFields.hasOwnProperty(field)) {
```
Comment (iamEAP) [inline src/steps/users/contact-field-equals.ts]: Hmm...  Not sure about the changes below.  My understanding is that user (contact) field data only ever shows up on the `dataFields` property (or at least that's what I'm seeing on [Iterable's API Docs](https://api.iterable.com/api/docs#users_getUser_0)

Wouldn't this prevent checking, e.g. `firstName` field?

Comment (saludangelito) [reply src/steps/users/contact-field-equals.ts]: image
Upon checking again. I did successfully remove email from the rest of the fields but its still being included in dataFields. So yeah I should revert back dataFields checking but remove other user field checking.

Comment (iamEAP) [reply src/steps/users/contact-field-equals.ts]: 👍

**Pull request merged**